File: code/backend/Odin/src/main/kotlin/pt/isel/odin/service/interfaces/ClassAttendanceService.kt
package pt.isel.odin.service.interfaces

import pt.isel.odin.model.ClassAttendance

/**
 * Service for class attendance.
 */
interface ClassAttendanceService {

    /**
     * Gets a class attendance by its id.
     *
     * @param id the class attendance id.
     */
    fun getById(id: Long): ClassAttendance

    /**
     * Gets all class attendances.
     */
    fun getAll(): List<ClassAttendance>

    /**
     * Saves a class attendance.
     *
     * @param classAttendRequest the class attendance info to save.
     */
    fun save(classAttendRequest: ClassAttendance): ClassAttendance

    /**
     * Updates a class attendance.
     *
     * @param classAttendRequest the class attendance info to update.
     */
    fun update(classAttendRequest: ClassAttendance): ClassAttendance

    /**
     * Deletes a class attendance by its id.
     *
     * @param id the class attendance id.
     */
    fun delete(id: Long)

    /**
     * Gets all class attendances by tech id.
     *
     * @param techId the tech id.
     */
    fun getByTechId(techId: Long): List<ClassAttendance>
}

File: code/frontend/Tyr/src/services/ClassAttendanceService.ts
import {makeApiRequest} from '../axios/apiRequest';
import {ClassAttendance} from "../model/ClassAtendance";

/**
 * Service to manage class attendances
 */
export class ClassAttendanceService {

    static basePath = '/classattendances';

    static async getById(id: number): Promise<ClassAttendance> {
        return makeApiRequest('get', `${(ClassAttendanceService.basePath)}/${id}`);
    }

    static async getAll(): Promise<ClassAttendance[]> {
        return makeApiRequest('get', ClassAttendanceService.basePath);
    }

    static async save(classAttendanceRequest: ClassAttendance): Promise<ClassAttendance> {
        return makeApiRequest('post', `${(ClassAttendanceService.basePath)}/save`, classAttendanceRequest);
    }

    static async update(classAttendanceRequest: ClassAttendance): Promise<ClassAttendance> {
        return makeApiRequest('put', `${(ClassAttendanceService.basePath)}/update`, classAttendanceRequest);
    }

    static async delete(id: number): Promise<void> {
        return makeApiRequest('delete', `${(ClassAttendanceService.basePath)}/${id}`);
    }
}